features/0351-purpose-decorator/README.md

```diff
@@ -0,0 +1,125 @@
+# Aries RFC 0351: Purpose Decorator
+- Authors: [Filip Burlacu](filip.burlacu@securekey.com)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2019-12-16
+- Status Note: implementation is being explored by SecureKey
+- Supersedes:
+- Start Date: 2018-12-05
+- Tags: feature, decorator
+
+## Summary
+
+This RFC allows Aries agents to serve as mediators or relays for applications that don't use DIDComm.
+It introduces:
+- A new decorator, the `~purpose` decorator, which defines the intent, usage, or contents of a message
+- A means for a recipient, who is not DIDComm-enabled, to register with an agent for messages with a particular purpose
+- A means for a sender, who is not DIDComm-enabled, to send messages with a given purpose through its agent to a target agent
+- Guidance for creating a protocol which uses the `~purpose` decorator to relay messages over DIDComm for non-DIDComm applications
+
+## Motivation
+
+This specification allows applications that aren't Aries agents to communicate JSON messages over DIDComm using Aries agents analogously to [mediators](/concepts/0046-mediators-and-relays/README.md). Any agent which implements this protocol can relay arbitrary new types of message for clients - without having to be updated and redeployed.
+
+## Tutorial
+
+> This RFC assumes familiarity with [mediators and relays](/concepts/0046-mediators-and-relays/README.md), [attachments](/concepts/0017-attachments/README.md), and [message threading](/concepts/0008-message-id-and-threading/README.md).
+
+### The `~purpose` Decorator
+
+The `~purpose` decorator is a JSON array which describes the *semantics* of a message - the role it plays within a protocol implemented using this RFC, for example, or the type or meaning of the data contained within.
+The purpose is the mechanism for determining which recipient(s) should be sent a message.
+
+Example: `"~purpose": ["team:01453", "delivery", "geotag", "cred"]`
+
+Each element of the purpose array is a string. An agent provides some means for recipients to *register* on a purpose, or class of purposes, by indicating the particular string values they are interested in.
+
+The particular registration semantics left to the implementation. Some possible examples include:
```

Comment:
How can an Agent differentiate between the intended semantics (if it supports more than one)?